src/storage/wal_replay.cpp

```diff
@@ -403,47 +404,86 @@ void ReplayState::ReplayDropTableMacro(BinaryDeserializer &deserializer) {
 // Replay Index
 //===--------------------------------------------------------------------===//
 void ReplayState::ReplayCreateIndex(BinaryDeserializer &deserializer) {
-	auto info = deserializer.ReadProperty<unique_ptr<CreateInfo>>(101, "index");
+
+	auto create_info = deserializer.ReadProperty<unique_ptr<CreateInfo>>(101, "index_catalog_entry");
+	auto index_info = deserializer.ReadProperty<IndexStorageInfo>(102, "index_storage_info");
+	D_ASSERT(index_info.IsValid());
+
+	auto &storage_manager = db.GetStorageManager();
+	auto &single_file_sm = storage_manager.Cast<SingleFileStorageManager>();
+	auto &block_manager = single_file_sm.block_manager;
+	auto &buffer_manager = block_manager->buffer_manager;
+
+	deserializer.ReadList(103, "index_storage", [&](Deserializer::List &list, idx_t i) {
+		auto &data_info = index_info.data_infos[i];
+		D_ASSERT(data_info.buffer_block_pointers.size() == index_info.buffer_counts[i]);
+
+		// read the data into buffer handles and convert them to blocks on disk
+		// then, update the block pointer
+		for (idx_t j = 0; j < data_info.buffer_block_pointers.size(); j++) {
```

Comment:
this can be a `auto& buffer_block_ptr`?

Reply:
I need `j` for another equally sized vector that contains the allocation sizes when reading the bytes: `list.ReadElement<bool>(data_ptr, data_info.buffer_allocation_sizes[j]);`